tests/testapps/AddressBookSample/AddressBook/ViewController.m

```diff
@@ -0,0 +1,135 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import "ViewController.h"
+#import "SingleContactViewController.h"
+#import "ViewAllContactsViewController.h"
+#import <AddressBook/ABAddressBook.h>
+
+@interface ViewController () <UITableViewDelegate>
+
+/**
+ * Displays an alert telling the user they must give the app permissions to use
+ * their AddressBook. Clears the menu so the user does not try to use other parts
+ * of the application.
+ */
+- (void)_accessDenied;
+
+@end
+
+@implementation ViewController {
+    NSArray* _menuOptions;
+}
+
+- (void)_accessDenied {
+    UIAlertController* alert = [UIAlertController alertControllerWithTitle:@"Cannot Use AddressBook"
+                                                                   message:@"You must give the app permission to use the AddressBook."
+                                                            preferredStyle:UIAlertControllerStyleAlert];
+    UIAlertAction* defaultAction = [UIAlertAction actionWithTitle:@"OK"
+                                                            style:UIAlertActionStyleDefault
+                                                          handler:^(UIAlertAction* action){
+                                                          }];
+    [alert addAction:defaultAction];
+    [self presentViewController:alert animated:YES completion:nil];
+
+    // Clear the menu to effectively disable the rest of the application
+    // (which is useless without AddressBook permissions).
+    _menuOptions = @[];
+    [self.tableView reloadData];
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    [self.tableView setHidden:YES];
+    _menuOptions = @[ @"View All Contacts", @"Create Contact", @"Modify Contact", @"Remove Contact" ];
```

Comment:
_menuOptions can be re-assigned to an empty array if the user doesn't grant any permissions.

---
In reply to: [70128241](https://github.com/Microsoft/WinObjC/pull/627#discussion_r70128241) [](ancestors = 70128241)